src/components/pluginManager.js

```diff
@@ -25,68 +25,85 @@ define(['events'], function (events) {
         this.pluginsList = [];
     }
 
-    PluginManager.prototype.loadPlugin = function (url) {
+    PluginManager.prototype.loadPlugin = function(pluginSpec) {
 
-        console.debug('Loading plugin: ' + url);
         var instance = this;
 
-        return new Promise(function (resolve, reject) {
+        function registerPlugin(plugin) {
+            instance.register(plugin);
 
-            require([url, 'globalize', 'appRouter'], function (pluginFactory, globalize, appRouter) {
-
-                var plugin = new pluginFactory();
-
-                // See if it's already installed
-                var existing = instance.pluginsList.filter(function (p) {
-                    return p.id === plugin.id;
-                })[0];
+            if (plugin.getRoutes) {
+                plugin.getRoutes().forEach(function (route) {
+                    definePluginRoute(instance, route, plugin);
+                });
+            }
+
+            if (plugin.type === 'skin') {
+
+                // translations won't be loaded for skins until needed
+                return Promise.resolve(plugin);
+            } else {
+                return new Promise((resolve, reject) => {
+                    loadStrings(plugin)
+                        .then(function () {
+                            resolve(plugin);
+                        })
+                        .catch(reject);
+                });
+            }
+        }
 
-                if (existing) {
-                    resolve(url);
-                    return;
-                }
+        if (typeof pluginSpec === "string") {
+            console.debug('Loading plugin: ' + pluginSpec);
```

Comment:
we should blurb that this is a deprecated method of registering a plugin here

Reply:
You mean like with a comment or so?

Reply:
no, I mean a message to browser console

Reply:
Can do that

Reply:
Fixed